calibr8/core.py

```diff
@@ -51,52 +51,111 @@ def __init__(
-        result = (
-            str(type(self))
-            + f"\n    ETI ({round(self.eti_prob, 3) * 100:.1f} %): [{round(self.eti_lower, 4)}, {round(self.eti_upper, 4)}] Δ={round(self.eti_width, 4)}"
-            + f"\n    HDI ({round(self.hdi_prob, 3) * 100:.1f} %): [{round(self.hdi_lower, 4)}, {round(self.hdi_upper, 4)}] Δ={round(self.hdi_width, 4)}"
-        )
-        return result
+    @property
+    def eti_x(self) -> numpy.ndarray:
+        """Values of the independent variable in the interval [eti_lower, eti_upper]"""
+        return self._eti_x
 
     @property
-    def eti_lower(self) -> float:
+    def eti_pdf(self) -> numpy.ndarray:
+        """Values of the posterior probability density at the positions `eti_x`."""
+        return self._eti_pdf
+
+    @property
+    def eti_lower(self) -> Union[float, numpy.ndarray]:
         """ Lower bound of the ETI. This is the first value in `eti_x`. """
-        return self.eti_x[0]
+        return self._eti_x[..., 0]
 
     @property
-    def eti_upper(self) -> float:
+    def eti_upper(self) -> Union[float, numpy.ndarray]:
         """ Upper bound of the ETI. This is the last value in `eti_x`. """
-        return self.eti_x[-1]
+        return self._eti_x[..., -1]
 
     @property
-    def eti_width(self) -> float:
+    def eti_width(self) -> Union[float, numpy.ndarray]:
         """ Width of the ETI. """
         return self.eti_upper - self.eti_lower
 
     @property
-    def hdi_lower(self) -> float:
+    def eti_prob(self) -> float:
+        """ Probability mass of the given equal tailed interval. """
+        return self._eti_prob
+
+    @property
+    def hdi_x(self) -> numpy.ndarray:
+        """Values of the independent variable in the interval [hdi_lower, hdi_upper]"""
+        return self._hdi_x
+
+    @property
+    def hdi_pdf(self) -> numpy.ndarray:
+        """Values of the posterior probability density at the positions `hdi_x`."""
+        return self._hdi_pdf
+
+    @property
+    def hdi_lower(self) -> Union[float, numpy.ndarray]:
         """ Lower bound of the HDI. This is the first value in `hdi_x`. """
-        return self.hdi_x[0]
+        return self._hdi_x[..., 0]
 
     @property
-    def hdi_upper(self) -> float:
+    def hdi_upper(self) -> Union[float, numpy.ndarray]:
         """ Upper bound of the HDI. This is the last value in `hdi_x`. """
-        return self.hdi_x[-1]
+        return self._hdi_x[..., -1]
 
     @property
-    def hdi_width(self) -> float:
+    def hdi_width(self) -> Union[float, numpy.ndarray]:
         """ Width of the HDI. """
         return self.hdi_upper - self.hdi_lower
 
+    @property
+    def hdi_prob(self) -> float:
+        """ Probability mass of the given highest density interval. """
+        return self._hdi_prob
+
+
+class UnivariateInferenceResult(InferenceResult):
+    """ The result of a numeric infer_independent operation with a univariate model. """
+    def __init__(
+        self,
+        median: float,
```

Comment:
Why is the median here and not in the base class already? You expect examples where the median is not useful?

Reply:
You can only do a marginal median, so there's no good measure for multivariate data. Taking all the marginal medians is not meaningful if the samples are, for example banana-shaped.
A banana-shape will be the inference result for a bivariate inference with a model that's linear in one and logistic in the other dimension, for example.

So that's why I think we should give a median only for the univariate case.

Reply:
Approved